ProfileManager.js

```diff
@@ -810,21 +976,59 @@ export default class ProfileManager {
       // signer for signing with the profileAgent's capability invocation key
       signer = new AsymmetricKey({
         capability: zcap,
-        invocationSigner: this.capabilityAgent.getSigner(),
+        invocationSigner: await this._getInvocationSigner(
+          {capabilityInvoker: 'local'}),
       });
-      this._cache.agentSigners.set(id, signer);
+      agentSignersCache.set(id, signer);
     }
     return signer;
   }
 
+  async _getInvocationSigner({profileAgentId, capabilityInvoker}) {
```

Comment:
Since I don't see an option to get a "profile" invocation signer, so this seems to be too ambiguous of a name. I think this should be `getAgentInvocationSigner` ... but I know we have `_getAgentSigner` already. Which, by the way, why doesn't this function call that if the type is `agent`? Maybe `_getAgentSigner` should just be modified to add this additional option for `local`?

That also leads me to ask -- is there a reason why we're caching `this.capabilityAgent.getSigner()`? Maybe we don't even need `_getInvocationSigner` at all and it's just a choice between `_getAgentSigner()` and `this.capabilityAgent.getSigner()`?

But that leads me to ask ... is there just one capability agent for all of these profile agents? That's a correlation risk. There should be a separate capability agent for each one.

Reply:
**Why do we have a single function?**

Creating a single function removes the need for having unnecessary branching logic in other parts of the code. I first started with the "a choice between `_getAgentSigner()` and `this.capabilityAgent.getSigner()`" then I threw it in the trash. The UX should be "give me a signer" and then your wish is granted. An example can be found here: https://github.com/digitalbazaar/bedrock-web-profile-manager/pull/33/files#diff-488c2e6bf3fd8c5d04f8af66519e12bfR512-R513

**Can we modify `_getAgentSigner`?**

I want to say no as it should have been deleted. No code references `_getAgentSigner`. The capabilityAgent signer is not a local profile agent signer. It's a signer created from an ephemeral `did:key`.

**...is there just one capability agent for all of these profile agents? That's a correlation risk.**

Yes. That is what I believe our currently deployed implementation does and I did not change that. Do we want to address this now or in a subsequent PR?

Reply:
> I want to say no as it should have been deleted. No code references _getAgentSigner. The capabilityAgent signer is not a local profile agent signer. It's a signer created from an ephemeral did:key.

If we're removing `_getAgentSigner` that's fine. I was confused because it still existed. It seems a bit odd that you can't get a profile signer from `getInvocationSigner` though. Also, clearly we're passing `profileAgentId` to `getInvocationSigner` ... so clearly we're going through a profile agent regardless... and that's not really captured in the name. We need to make some kind of adjustment here, I guarantee other people will be confused.

> Yes. That is what I believe our currently deployed implementation does and I did not change that. Do we want to address this now or in a subsequent PR?

We need to address it now because we're introducing the correlation risk ... that wasn't there before.

Reply:
> If we're removing `_getAgentSigner` that's fine. I was confused because it still existed. It seems a bit odd that you can't get a profile signer from `getInvocationSigner` though. Also, clearly we're passing `profileAgentId` to `getInvocationSigner` ... so clearly we're going through a profile agent regardless... and that's not really captured in the name. We need to make some kind of adjustment here, I guarantee other people will be confused.

Understood. To be clear `profileAgentId` is not required to get the local signer but that may change in order to mitigate some security concerns. It seems as if we can just make this function handle getting a profile signer and be done with it. Any objections in proceeding with that?

> We need to address it now because we're introducing the correlation risk ... that wasn't there before.

I see the issue now. We were invoking capabilities for profiles to a profile agent. We are now delegating all profile agent's capabilities to a single ephemeral `did:key` signer creating the correlation risk. We need to maintain our 1:1 mapping from profile agent to ephemeral `did:key` signer so that we stay on par with the security profile we had before. Is that on track?

Reply:
> It seems as if we can just make this function handle getting a profile signer and be done with it. Any objections in proceeding with that?

Just getting a **profile** signer? Is that a typo? If not, I'm more confused.

> We need to maintain our 1:1 mapping from profile agent to ephemeral did:key signer so that we stay on par with the security profile we had before. Is that on track?

That's right.

Reply:
> Just getting a **profile** signer? Is that a typo? If not, I'm more confused.

The excerpt, "just make this function handle getting a profile signer ", means make it so that the function can also return a profile signer. I am **not** suggesting only returning a profile signer.

Reply:
> The excerpt, "just make this function handle getting a profile signer ", means make it so that the function can also return a profile signer. I am not suggesting only returning a profile signer.

Ah! Ok. :)

Reply:
I opted to rename the function to `getAgentSigner()` and it takes the option `profileAgentId` and `useEphemeralSigner`.